.github/workflows/build.yaml

```diff
@@ -10,56 +9,58 @@ on:
     branches: 
       - '**'
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest] #FIXME windows-latest fails to run tests
+        os: [macos-latest, ubuntu-latest]
 
     steps:
-      # Checkout code
       - uses: actions/checkout@v4
 
-      # Set up Node
       - name: Use Node 20
         uses: actions/setup-node@v4
         with:
           node-version: 20
 
-      # Run install dependencies
       - name: Install dependencies
         run: npm run install:ci
 
-      # Build extension
       - name: Run build
         run: npm run package
 
-      # Generate .vsix package for the extension 
       - name: Generate .vsix package
         if: runner.os == 'Linux'  
         run: |
           npm install -g @vscode/vsce
           vsce package
 
-      # Upload VSIX artifact 
       - name: Upload VSIX artifact
         if: runner.os == 'Linux'  
         uses: actions/upload-artifact@v4
         with:
-          name: vscode-granite-vsix-${{ runner.os }}-${{ github.run_id }}  # Unique name with OS and run ID
+          name: vscode-granite-vsix-${{ runner.os }}-${{ github.run_id }}
           path: "./vscode-granite-*.vsix"
 
-      # Run tests
-      - name: Run Test
-        uses: coactions/setup-xvfb@b6b4fcfb9f5a895edadc3bc76318fae0ac17c8b3 #v1.0.1
-        with:
-          run: npm test
+      - name: Run Test on Ubuntu
+        if: runner.os == 'Linux'
+        run: |
+          # Install Xvfb
+          sudo apt-get update
+          sudo apt-get install -y xvfb
+          # Start Xvfb
+          Xvfb :99 -screen 0 1920x1080x24 &
+          export DISPLAY=:99
+          # Run your tests
+          npm test
```

Comment:
whole setup is unnecessary, as xvfb is available on ubuntu runners now. 

See: 

- https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#installed-apt-packages
- https://github.com/coactions/setup-xvfb/issues/21#issuecomment-2021625936